src/datasources/system/components/SystemQueryEditor.tsx

```diff
@@ -13,13 +13,26 @@ export function SystemQueryEditor({ query, onChange, onRunQuery }: Props) {
     onRunQuery();
   }
 
+  const onIdBlur = (e: React.FocusEvent<HTMLInputElement>) => {
+    const id = e.currentTarget.value;
+    if (query.systemId !== id) {
+      onChange({ ...query, systemId: id }) //change to systemid
+      onRunQuery(); 
+    }
+  }
+
   return (
     <div>
       <InlineFieldRow >
         <InlineField label="Query type">
           <RadioButtonGroup options={enumToOptions(QueryType)} onChange={onQueryTypeChange} value={query.queryKind} />
         </InlineField>
       </InlineFieldRow>
+      <InlineFieldRow>
+        <InlineField label="Minion ID">
+          <Input placeholder="All systems" onBlur={onIdBlur} />
```

Comment:
We need to set the `defaultValue` prop here to the value in the query so it's filled in when the user re-opens the query editor.